### PR TITLE
Allow using the public IP for SSH connections when the private network is enabled

### PR DESCRIPTION
**Pull request by NotANormalNerd**
Adds possibility to use private network first and have a global rancher instance, outside of the network scope.

### CODE REVIEW

File: driver.go
Comment (mxschmitt) [inline driver.go:401]:
```suggestion
	return d.GetIP()
```

**Review by mxschmitt — APPROVED** [review]
LGTM, but let's wait for Jonas's feedback.

**Comment by NotANormalNerd** [comment]
I have finally found a solution in combination with the hetzner cloud controller. https://github.com/hetznercloud/hcloud-cloud-controller-manager/pull/144

**Comment by JonasProgrammer** [comment]
Hi there and sorry, I was rather busy the last days and could look into this only now.

The code itself LGTM, I'd be happy to merge it, if you do have a use case for that. Now that you have closed it, I assumed your issue is resolved. But then again, the PR you linked was denied.

So is there still a need for this? Perhaps we can still discuss this in the corresponding issue.
The problem is, when it comes to K8s/Rancher, I'm completely out of the loop, so I cannot really tune in. I'm open to new features, if they have valid use cases, but I also try to keep the concerns of a machine driver and an orchestrator on a higher level separated.

**Comment by NotANormalNerd** [comment]
Hello Jonas,
No the PR is not needed anymore, since everything works fine when the K8s is managed by the hetzner-cloud-controller-manager, but you have to find the correct combination of config flags.

The issue with the linked PR is for JKammerer, that they want to manage the private networks, but we already do this here on deploy. So we have to disable that, but allow the manager to recoginze and setup the private network in K8s.

**Pull request closed without merging**
